src/snowflake/cli/api/project/definition.py

```diff
@@ -90,3 +96,20 @@ def default_role():
 def default_application(project_name: str):
     user = sanitize_identifier(get_env_username() or DEFAULT_USERNAME).lower()
     return append_to_identifier(to_identifier(project_name), f"_{user}")
+
+
+def _no_duplicates_constructor(loader, node, deep=False):
+    """
+    Raises error it there are duplicated keys on the same level in the yaml file
+    """
+    mapping = {}
+
+    for key_node, value_node in node.value:
+        key = loader.construct_object(key_node, deep=deep)
+        value = loader.construct_object(value_node, deep=deep)
+        if key in mapping.keys():
+            raise ClickException(
+                f"While loading the project definition file, duplicate key was found: {key}"
```

Comment:
Is it possible to provide a full path? Like `Found duplicated key "entities.test_procedure.handler" while loading project definition file`?

Reply:
The custom constructor  is used to construct a single tag, so it has no info about parent nodes.
I can add some variable to track it, but AFAIK it would mean we have to traverse all the nodes while loading each tag, and that would significantly increase time complexity of the loader function